File: baselibrary/src/main/java/com/anningtex/baselibrary/base/BasePresenter.java
package com.anningtex.baselibrary.base;

/**
 * @ClassName: BasePresenter
 * @Description: java类作用描述
 * @Author: alvis
 * @CreateDate: 2020/6/19 16:49
 */
public class BasePresenter<V extends BaseContract.View> implements BaseContract.Presenter<V> {
    private V view;

    @Override
    public void attachView(V view) {
        this.view = view;
    }

    @Override
    public void detachView() {
        view = null;
    }

    public boolean isViewAttached() {
        return view != null;
    }

    public V getAttachView() {
        return view;
    }
}

File: baselibrary/src/main/java/com/anningtex/baselibrary/base/BasePresenter.kt
package com.anningtex.baselibrary.base

open class BasePresenter<V : BaseContract.View?> : BaseContract.Presenter<V> {
    var attachView: V? = null
        private set

    override fun attachView(view: V) {
        attachView = view
    }

    override fun detachView() {
        attachView = null
    }

    val isViewAttached: Boolean get() = attachView != null

}